# Matplotlib - Convert Image File to Numpy Array

In [6]:
from matplotlib.image import imread

Use the imread() function of the image module of matplotlib to convert the image file below to 3D numpy array (x-pos,y-pos,color-array). Which reads the colour code of each pixel and converts the rgb channels into array

<img src="./assets/images/ds_framework.png" />

In [7]:
ds_framwork_screenshot_data = imread("./assets/images/ds_framework.png")

Lets preview the imported image data

In [10]:
ds_framwork_screenshot_data.dtype

dtype('float32')

In [11]:
ds_framwork_screenshot_data.shape

(596, 1053, 4)

From the shape above, we can see that its a 1053 x 596px image with four color channels r,g,b,a

In [15]:
ds_framwork_screenshot_data[0]

array([[0.08627451, 0.08627451, 0.11372549, 1.        ],
       [0.08627451, 0.08627451, 0.11372549, 1.        ],
       [0.32156864, 0.32156864, 0.3372549 , 1.        ],
       ...,
       [0.99607843, 0.99607843, 0.99607843, 1.        ],
       [0.9647059 , 0.9647059 , 0.9647059 , 1.        ],
       [0.08627451, 0.08627451, 0.11372549, 1.        ]],
      shape=(1053, 4), dtype=float32)

When we access the array at root index 0 => it returns the 2D array containing the arrays of pixels